seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/excel/ExcelReaderListener.java

```diff
@@ -0,0 +1,277 @@
+        for (int i = 0; i < headMap.size(); i++) {
+            String header = headMap.get(i).getStringValue();
+            if (!"null".equals(header)) {
+                customHeaders.put(i, header);
+            }
+        }
+    }
+
+    @Override
+    public void invoke(Map<Integer, Object> data, AnalysisContext context) {
+        cellCount = data.size();
+        SeaTunnelRow seaTunnelRow = new SeaTunnelRow(fieldTypes.length);
+        Map<Integer, Cell> cellMap = context.readRowHolder().getCellMap();
+        int i = 0;
+        for (; i < fieldTypes.length; i++) {
+            if (cellMap.get(i) == null) {
+                seaTunnelRow.setField(i, null);
+            } else {
+                Object cell = convert(data.get(i), cellMap.get(i), fieldTypes[i]);
+                seaTunnelRow.setField(i, cell);
+            }
+        }
+        seaTunnelRow.setTableId(tableId);
+        output.collect(seaTunnelRow);
+    }
+
+    @Override
+    public void doAfterAllAnalysed(AnalysisContext context) {
+        log.info("excel parsing completed");
+    }
+
+    @Override
+    public void onException(Exception exception, AnalysisContext context) {
+        log.error("cell parsing exception :{}", exception.getMessage());
+        if (exception instanceof ExcelDataConvertException) {
+            ExcelDataConvertException excelDataConvertException =
+                    (ExcelDataConvertException) exception;
+            log.error(
+                    "row:{},cell:{},data:{}",
+                    excelDataConvertException.getRowIndex(),
+                    excelDataConvertException.getColumnIndex(),
+                    excelDataConvertException.getCellData());
+        }
+    }
+
+    private String getCellValue(ReadCellData cellData) {
+
+        if (cellData.getStringValue() != null) {
+            return cellData.getStringValue();
+        } else if (cellData.getNumberValue() != null) {
+            return cellData.getNumberValue().toString();
+        } else if (cellData.getOriginalNumberValue() != null) {
+            return cellData.getOriginalNumberValue().toString();
+        } else if (cellData.getBooleanValue() != null) {
+            return cellData.getBooleanValue().toString();
+        } else if (cellData.getType() == CellDataTypeEnum.EMPTY) {
+            return "";
+        }
+        return null;
+    }
+
+    @SneakyThrows(JsonProcessingException.class)
+    private Object convert(Object field, Cell cellRaw, SeaTunnelDataType<?> fieldType) {
+
+        if (cellRaw == null && field == null) {
+            return null;
+        }
+        String cellValue = (String) field;
+        ReadCellData cellData = (ReadCellData) cellRaw;
+        if (cellRaw != null) {
+            cellValue = getCellValue(cellData);
+        }
+        SqlType sqlType = fieldType.getSqlType();
+
+        if (cellValue == null || (cellValue.equals("") && sqlType != SqlType.STRING)) {
+            return null;
+        }
+
+        switch (sqlType) {
```

Comment:
can we reuse https://github.com/apache/seatunnel/pull/8064/files#diff-f7eb796028f745bd57298e931d7712c0105c5e1e09fb6679fd7c07bb2febe211R243 ?

Reply:
> can we reuse https://github.com/apache/seatunnel/pull/8064/files#diff-f7eb796028f745bd57298e931d7712c0105c5e1e09fb6679fd7c07bb2febe211R243 ?

I'll try